ktor-client/ktor-client-core/common/src/io/ktor/client/engine/HttpClientEngine.kt

```diff
@@ -45,23 +52,24 @@ interface HttpClientEngine : CoroutineScope, Closeable {
      */
     @InternalAPI
     fun install(client: HttpClient) {
+        // This interceptor should be placed after HttpRequestLifecycle and HttpTimeout features.
```

Comment:
Consider adding a new phase instead

Reply:
It would be nice to have a test with the sequence:
 1. `HttpRequestBuilder` created without a client
 2.  Fire and fail it(with 404)
 3. Fix an issue in the builder
 4. Try again(with 200 OK)

Reply:
Or we should consider deprecating `1`

Reply:
Done, I've added the test.